### ISSUE TITLE
Dropdown View not Displaying

### ISSUE THREAD

**Issue by goonerDroid** [lolobosse/ContactsAutoCompleteTextView#8]
Saw this weird bug where the dropdown view does not render when the Textview is selected.But writing the contact name and after some delay(10-15 mins) it shows the dropdown view.Noticed this weird bug on Lollipop devices although it works fine in pre-Lollipop devices.

PS.Also encountered this bug on Playstore sample app as well.

**Comment by lolobosse** [comment]
Good morning @goonerDroid,

I do not understand what could be the reason of this bug but I'll have a look at it. Therefore could you give me roughly the number of your contacts and the option you selected (mail or phone). What do you have for phone?

The delay can be caused because it is asynchronous: the dropdown comes only when the data processing is finished.

**Comment by goonerDroid** [comment]
Tried searching multiple contacts as well as single contact(phone).Their is a considerable amount of delay, after some time it crashes randomly.Testing on Moto G

**Comment by goonerDroid** [comment]
@lolobosse is there any way we can optimize the Search thread in the Custom adapter class??and btw it is not OS specific issue,it is directly proportional  to the number of contact result

**Comment by lolobosse** [comment]
Yeah that's what I thought, I'll have a look but you can have a look at the search thread if you want. I'll be happy for any suggestion. The data are retrieved in `CustomAdapter` and the filter is done in `AutoCompleteTextView`

**Comment by goonerDroid** [comment]
Tried loading with async also.Same results!

**Comment by lolobosse** [comment]
I discussed that with a friend (@matthieubulte) and he meant that there is no that much to do against that but a bit of profiling and see if there is a better data structure to deal with that problem (list is not that efficient).

Another way could be to make a call to the Android Contact DB as you type.

I need some time to solve that (and do not have that much: I'm studying and building my startup!)

**Comment by goonerDroid** [comment]
Cool.Me too will will be working on it.Will let you know if there is progress.Anyways best of luck for your startup!

**Comment by lolobosse** [comment]
Can you test it again? I almost rewrote the whole lib (not on master branch but the one I referenced the commit from)

**Comment by goonerDroid** [comment]
Same results buddy!!The device i'm testing has approx 3000 contacts.Has a delay of 1-2 mins!

**Comment by lolobosse** [comment]
You tested on the test branch right ^^?

**Comment by goonerDroid** [comment]
Tested on the properCustomAdapter branch!Still the same results.Can we pre-load all the contacts and cache them before they are requested??You are doing that i guess?

**Comment by lolobosse** [comment]
Yes, they're cached... Do you have the same problem on Google contacts search? Or is it working fine?

**Comment by goonerDroid** [comment]
Why aren't  you using Loader Callbacks?

**Comment by lolobosse** [comment]
Because I didn't know they exist, I made Callbacks myself but it seems to be a good idea. Do you have a API ref about that topic? :smile: :+1:

**Comment by goonerDroid** [comment]
checkout his implementation of Loaders and weak references in Asyncs https://github.com/codinguser/android_contact_picker. Trying to hack these two together!

**Comment by lolobosse** [comment]
Ok ok, I see, I'll implement that asap. That's a good idea, I started 4 years ago and these things didn't exist but it's a very good idea :+1: :+1: :stuck_out_tongue_winking_eye:

**Comment by goonerDroid** [comment]
Cool man even i' m doing it now!Let's close this goddamn issue!!

**Comment by lolobosse** [comment]
I'm at work right now, so if you want to make a pull request, I'll be very happy to look at it :smile: 
If you want, @alexjlockwood did a great post about that topic: http://www.androiddesignpatterns.com/2012/07/loaders-and-loadermanager-background.html

**Comment by lolobosse** [comment]
So I'm back on that project because I have a bit more time now.

I'll try the following:
1. Implement the a `LoaderManager` in the adapter.
2. Find a way to get 10K contacts on a device to be able to test it properly.
3. Remove the fake async (it waits to have all the results). Let's try to stream the results.
